contracts/finance/andromeda-rate-limiting-withdrawals/src/contract.rs

```diff
@@ -0,0 +1,578 @@
+use crate::state::{ACCOUNTS, ALLOWED_COIN};
+use ado_base::ADOContract;
+use andromeda_finance::rate_limiting_withdrawals::{
+    AccountDetails, CoinAllowance, ExecuteMsg, InstantiateMsg, MigrateMsg, QueryMsg,
+};
+use common::{
+    ado_base::{AndromedaQuery, InstantiateMsg as BaseInstantiateMsg},
+    encode_binary,
+    error::ContractError,
+    require,
+};
+use cosmwasm_std::{
+    entry_point, BankMsg, Binary, Coin, CosmosMsg, Deps, DepsMut, Env, MessageInfo, Response,
+    StdError, Uint128,
+};
+use cw2::{get_contract_version, set_contract_version};
+
+use cw_utils::{nonpayable, one_coin};
+use semver::Version;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:andromeda-splitter";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[entry_point]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    ALLOWED_COIN.save(deps.storage, &msg.allowed_coin)?;
+
+    ADOContract::default().instantiate(
+        deps.storage,
+        env,
+        deps.api,
+        info,
+        BaseInstantiateMsg {
+            ado_type: "rate-limiting-withdrawals".to_string(),
+            ado_version: CONTRACT_VERSION.to_string(),
+            modules: msg.modules,
+            primitive_contract: None,
+        },
+    )
+}
+
+#[entry_point]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::Deposit { recipient } => execute_deposit(deps, env, info, recipient),
```

Comment:
May be useful to have another message to update allowed coins restricted to owner/operator